### PR TITLE
Always store and compare the email address as lower case

### PR DESCRIPTION
**Pull request by nickvergessen**
1. Needs a database query executed on updating:
    ```sql
    UPDATE `oc_preferences` SET `configvalue` = LOWER(`configvalue`) WHERE `appid` = 'settings' AND `configkey` = 'email';
    ```
2. Before merging into master, are the details stored somewhere else ❓ 
3. Add a partial index for the first ~80 characters ❓ 
4. Bump version number to trigger the migration

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

File: lib/private/AllConfig.php [hunk 1]
```diff
@@ -260,6 +260,10 @@ public function setUserValue($userId, $appName, $key, $value, $preCondition = nu
 		// TODO - FIXME
 		$this->fixDIInit();
 
+		if ($appName === 'core' && $key === 'email') {
```
Comment (CarlSchwan) [inline lib/private/AllConfig.php:263]:
```suggestion
		if ($appName === 'core' && $key === 'email' && is_string($value)) {
```

Should be enough to tell psalm to be more quite :)

Comment (nickvergessen) [reply lib/private/AllConfig.php:263]: Added a cast instead

**Review by CarlSchwan — APPROVED** [review]

**Comment by Pytal** [comment]
> 4. Bump version number to trigger the migration
Needs a version bump

**Review by skjnldsv — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

**Comment by nickvergessen** [comment]
> Please run: bash build/autoloaderchecker.sh
> And commit the result

Can you do it?

**Comment by Pytal** [comment]
⚙ failure unrelated

**Pull request merged**